tapfreighter/chain_porter.go

```diff
@@ -870,11 +870,45 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 
 	// If we have a non-interactive proof, then we'll launch several
 	// goroutines to deliver the proof(s) to the receiver(s).
-	err := fn.ParSlice(ctx, pkg.OutboundPkg.Outputs, deliver)
+	instanceErrors, err := fn.ParSliceErrCollect(
+		ctx, pkg.OutboundPkg.Outputs, deliver,
+	)
 	if err != nil {
 		return fmt.Errorf("error delivering proof(s): %w", err)
 	}
 
+	// If there were any errors during the proof delivery process, we'll
+	// report them here and return the first error we encountered.
+	if len(instanceErrors) > 0 {
+		var firstErr error
+		for idx := range instanceErrors {
```

Comment:
`range` over a map will be unordered:

https://go.dev/blog/maps

But you could build a slice of ordered keys with something like:

`slices.Sort(maps.Keys(instanceErrors))`

IIUC you want to log all errors but only return the first one.

Reply:
I've added `fn.PeekMap` to select on of the errors. It doesn't select deterministically but I think that's OK in this case.